ecommerce/api_test.py

```diff
@@ -238,32 +245,32 @@ def test_less_or_equal_to_zero(self):
 
             assert Order.objects.count() == 0
 
-    def test_create_order(self):
+    @ddt.data(True, False)
+    def test_create_order(self, has_coupon):
         """
         Create Order from a purchasable course
         """
         course_run, user = create_purchasable_course_run()
         discounted_price = round(course_run.courseprice_set.get(is_valid=True).price/2, 2)
-        price_dict = {
-            "price": discounted_price
-        }
+        coupon = None
+        if has_coupon:
+            coupon = CouponFactory.create(content_object=course_run)
+        price_tuple = (discounted_price, coupon)
 
         with patch(
             'ecommerce.api.get_purchasable_course_run',
             autospec=True,
             return_value=course_run,
         ) as get_purchasable, patch(
-            'ecommerce.api.get_formatted_course_price',
+            'ecommerce.api.calculate_run_price',
             autospec=True,
-            return_value=price_dict,
-        ) as get_price:
+            return_value=price_tuple,
+        ) as _calculate_run_price:
```

Comment:
general question: why are we patching this function? is it really difficult for us to manually set up the data to make this work? as it is, this is another patch that makes the test more difficult to understand and would fail on a refactor that keeps the exact same behavior but changed implementation